scripts/heratape_backup.py

```diff
@@ -0,0 +1,330 @@
+        for tape in empty_tapes:
+            if tape['tape_id'] == usetape_id:
+                return tape['slot'], tape['tape_id'], tape['usage'] 
+        return None, None, None # if no tapes found, its probably time for a fresh batch of tapes!
+        
+    #Armed with all our metadata from the Librarian, we can move to the Tape side of things
+    
+    # prepare tape for backup
+    drivetapeid = tape_in_drive(mydrive)
+    logger.info(f'Drive {mydrive}, Tape {drivetapeid}')
+    #check the total against the remaining space
+    tape_usage = query_tape_usage(drivetapeid,testing=TESTING)
+    if tape_usage >0:
+        logger.info(f'Size of this backup = {np.sum(sizes)/1e12} TB is < the {(TAPESIZE - tape_usage)/1e12}TB theoretically remaining on the tape')
+        logger.info('Load new tape')
+   
+        #load a fresh tape
+        #1 find an empty slot.  If there isn't one, we're done. Be ok if theres more than one
+        empty_slots = find_empty_slots()
+        if len(empty_slots)==0: 
+            logger.info("ERROR: No empty slot into which I can unload a tape. The jukebox is too full! Remove a tape and try again.")
+            sys.exit()
+        #if theres more than one, choose the topmost empty slot
+        if len(empty_slots)==1:
+            emptyslot = empty_slots
+        else:
+            emptyslot = empty_slots[0]
+        #unload working drive to this empty slot
+        logger.info(f'unloading drive {mydrive} back to slot {empyslot}') 
+        unload_tape(mydrive,emptyslot)
+        logger.info('tape unloaded')
+        #select a fresh tape
+        #  this might require human intervention, which we pessimistically assume might be the case until 
+        #  proven otherwise
+        waitingforhuman = True
+        while(waitingforhuman):
+            update_tapes(testing=TESTING) #check for any new tapes in the library
+            newslot,newtape_id,newtape_usage = select_empty_tape(mydrive,testing=TESTING)
+            if newslot is None:
+                if mydrive==0:
+                    logger.warning("ALERT: New Tapes needed in slots 13-24")
+                    #sys.exit()#TODO: pause and wait like a good daemon
+                    time.sleep(30)
+                    continue
+                else:
+                    logger.warning("ALERT: New Tapes needed in slots 1-12")
+                    #sys.exit()  #TODO: pause ad wait like a good daemon
+                    time.sleep(30)
+                    continue
+            else:
+                #we have an empty tape lets proceed
+                logger.info(f'loading {newtape_id} from {newslot} to drive {mydrive}')
+                load_tape(mydrive,newslot)
+                logger.info('tape load complete')
+                drivetapeid = newtape_id
+                break
+    if True:
+        #write out the full file list to a txt for offline testing
+        F = open('backup.txt','w')
```

Comment:
This is going to be overwritten constantly. Do you want to update the file name?